gateway/src/apicast/policy/rate_limit/rate_limit.lua

```diff
@@ -133,9 +134,9 @@ local function build_limiters_and_keys(type, limiters, redis, error_settings, co
     local key = limiter.template_string:render(
       ngx_variable.available_context(context))
     if limiter.key.scope == "global" then
-      key = format("%s_%s", type, key)
+      key = format("%d_%s_%s", timestamp + window, type, key)
```

Comment:
You can't use just timestamp + window. In case of multiple gateways they all need to generate the same key. That can be achieved by generating a time bucket.

The formula is: `time - (time % window)`. That will for the whole time of `window` return the same value and then flip to the next one when the `window` is over.

Reply:
Fixed.